math/classical_sieve_of_eratosthenes.cpp

```diff
@@ -1,21 +1,20 @@
-/*
+/**
  * Classical Sieve of Eratosthenes is an algorithm to find the primes
  * that is between 2 to N (as defined in main) in fastest way.
  *
  * Time Complexity  : O(N * log(log(N)))
  * Space Complexity : O(N)
- */
+**/
 
 #include <iostream>
+#include <climits>
 
-#define MAX 10000000
+bool isprime[INT_MAX];
```

Comment:
use dynamic memory allocation
try to allocate the memory using the `new` keyword instead of using `INT_MAX`
and also deallocate the memory after use using `delete` keyword

if you wish to allocate a bool array of size 20 (say) then you should write
`bool *arr = new bool[20];`
and to delete this allocated memory write
`delete [] arr;`

i hope now things are clear

Reply:
To add to this, `INT_MAX` bool array will try to allocate **2GB** of ram!! use only as much needed

Reply:
@kvedala can you help me how to implement this dynamically?